### PR TITLE
Update Microsoft.AspNetCore.Hosting version for .netcoreapp3.1+

### PR DESCRIPTION
**Pull request by mrstork**
## Changes

- Use `Microsoft.AspNetCore.App` framework reference for versions `netcoreapp3.1` and up
- Remove direct dependency on Microsoft.AspNetCore.Http for versions `netcoreapp3.1` and up
- Maintain the same package versions for `netstandard2.0` and `netstandard2.1`

The changes here follow the [offical guide of how to support multiple aspnet core versions](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-6.0&tabs=visual-studio#support-multiple-aspnet-core-versions)

### Checklist
- [x] I ran Unit Tests locally.
- [x] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile locally using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj [hunk 1]
```diff
@@ -33,33 +33,10 @@
     <ProjectReference Include="..\..\..\WEB\Src\EventCounterCollector\EventCounterCollector\EventCounterCollector.csproj" />
     <ProjectReference Include="..\..\..\LOGGING\src\ILogger\ILogger.csproj" />
     
-    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.1.1" />
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.2.7" />
```
Comment (cijothomas) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj:36]: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj We should borrow whats done in OpenTelemetry for similar needs.

Comment (mrstork) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj]: I also took this to mean we don't want to remove support for `netstandard2.0` just yet and I should just update 3.1+. If that's not what you expected please let me know, happy to update further.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
     <AssemblyName>Microsoft.ApplicationInsights.AspNetCore</AssemblyName>
-    <TargetFrameworks>netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
 
     <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netstandard2.0</TargetFrameworks>
```
Comment (mrstork) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj:8]: Not sure how to handle two target frameworks entries

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj [hunk 3]
```diff
@@ -33,33 +33,49 @@
...
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.1'">
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.1.1" />
+    <PackageReference Include="Microsoft.AspNetCore.Http" Version="2.1.22" />
+    <PackageReference Include="System.Text.Encodings.Web" Version="4.5.1" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.2.7" />
```
[the remark this answers is not in the document]
Comment (mrstork) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj:68]: Does that look like this?
```
<FrameworkReference Include="Microsoft.AspNetCore.Hosting" />
```

[the remark this answers is not in the document]
Comment (Romfos) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj:68]: > Does that look like this?
> 
> ```
> <FrameworkReference Include="Microsoft.AspNetCore.Hosting" />
> ```

<FrameworkReference Include="Microsoft.AspNetCore.App" />

[the remark this answers is not in the document]
Comment (Romfos) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj:68]: <FrameworkReference Include="Microsoft.AspNetCore.App" />

[the remark this answers is not in the document]
Comment (Romfos) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj:68]: No, it is okay. Final app will have dependencies only for used libraries

**Review by Romfos — APPROVED** [review]

**Comment by cijothomas** [comment]
The change to the .csproj looks good. Please remove the whitespace changes in changelog.md file, and we are good to merge.

**Comment by mrstork** [comment]
Thanks for all your help @cijothomas and @Romfos !

**Review by cijothomas — APPROVED** [review]
LGTM.

**Comment by cijothomas** [comment]
Please update description to say we are following the offiical guidance : https://docs.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-6.0&tabs=visual-studio#support-multiple-aspnet-core-versions

**Comment by mrstork** [comment]
@cijothomas/@Romfos - Is there anything outstanding on this work or anything I can do to help merge this?
My intention is not to rush anything, looking to find out if there's something I should still do.

**Comment by sorensenmatias** [comment]
It would be great to get this merged - we are fighting the old dependencies brought in by Microsoft.AspNetCore.Hosting 2.1.1 when targeting .NET core.
Can I help with anything?

**Pull request closed without merging**